File: happyShopping_manager_web/src/main/webapp/js/service/brandService.js
//自定义服务层
app.service('brandService', function ($http) {
    //查询品牌列表
    this.findAll = function () {
        return $http.get('../brand/list.action');
    };

    //分页查看品牌列表
    this.listByPage = function (pageNum, pageSize) {
        return $http.get('../brand/listByPage.action?pageNum=' + pageNum + '&pageSize=' + pageSize);
    };

    //增加方法
    this.add = function (entity) {
        return $http.post('../brand/add.action', entity);
    };

    //修改方法
    this.update = function (entity) {
        return $http.post('../brand/update.action', entity);
    };

    //通过id找到brand的信息
    this.findBrandById = function (id) {
        return $http.get('../brand/findBrandById.action?id=' + id);
    };

    //批量删除
    this.dele = function (selectIds) {
        return $http.get('../brand/delete.action?ids=' + selectIds)
    };
    //条件查询
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../brand/search.action?pageNum=' + pageNum + '&pageSize=' + pageSize, searchEntity);
    };
    //下拉列表品牌查询
    this.findBrandList = function () {
        return $http.get('../brand/selectOptionList.action');
    }
});

File: happyShopping_sellergoods_interface/src/main/java/com/happyShopping/sellergoods/service/BrandService.java
package com.happyShopping.sellergoods.service;

import com.happyShopping.model.Brand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 返回所有品牌
     * @return
     */
    public List<Brand> getAll();

    /**
     * 分页显示查看品牌
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult getBrandsByPage(int pageNum, int pageSize);

    /**
     * 增加品牌
     * @param brand
     */
    public void add(Brand brand);

    /**
     * 修改品牌信息
     * @param brand
     */
    public void update(Brand brand);

    /**
     * 通过id得到品牌
     * @param id
     */
    public Brand findBrandById(Long id);

    /**
     * 通过id删除
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 条件查询
     * @param brand
     * @param pageNum
     * @param pageSize
     */
    public PageResult search(Brand brand,int pageNum,int pageSize);

    /**
     * 类型模板的品牌下拉列表的数据
     * @return
     */
    List<Map> selectOptionList();
}
